eq-author-api/schema/tests/subscription.test.js

```diff
@@ -157,6 +175,64 @@ describe("subscriptions", () => {
         title: "",
       });
 
+      const p = iterator.next();
+      const result = await Promise.race([p, wait(50)]);
+      expect(result).toBe("timeout");
+    });
+  });
+  describe("publishStatusUpdated", () => {
+    const publishStatusSubscription = gql`
+      subscription PublishStatus($id: ID!) {
+        publishStatusUpdated(id: $id) {
+          id
+          publishStatus
+        }
+      }
+    `;
+
+    it("should send event when publish status goes from published to unpublished", async () => {
+      ctx = await buildContext(
+        {
+          publishStatus: PUBLISHED,
+          sections: [
+            {
+              pages: [
+                {
+                  title: "some value",
+                },
+              ],
+            },
+          ],
+        },
+        { admin: true }
+      );
+      questionnaire = ctx.questionnaire;
+      iterator = await executeSubscription(publishStatusSubscription, {
+        id: questionnaire.id,
+      });
+      const pageId = questionnaire.sections[0].pages[0].id;
+      await updateQuestionPage(ctx, {
+        id: pageId,
+        title: "asd",
+      });
+      const result = await iterator.next();
+      expect(result.value.data.publishStatusUpdated.publishStatus).toBe(
+        UNPUBLISHED
+      );
+    });
+    it("should not send updates unless status goes from published to unpublished", async () => {
+      ctx = await buildContext({ publishStatus: UNPUBLISHED });
+      questionnaire = ctx.questionnaire;
+
+      iterator = await executeSubscription(publishStatusSubscription, {
+        id: questionnaire.id,
+      });
+
+      await updateQuestionnaire(ctx, {
+        id: questionnaire.id,
+        publishStatus: AWAITING_APPROVAL,
+      });
+
       const p = iterator.next();
```

Comment:
If the variable above is changed, maybe do this one at the same time.

Reply:
I agree, one letter vars is not nice... But tbh I dont know what to call this, not sure what it does really. Just copied it from the above tests. I could call it something like `nextIterator` or `iteratorNext`?

Reply:
I have renamed to `iteratorNext`